src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -70,16 +74,16 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 
 		if (connectionCompletionOptions) {
 			// Somehow, connectionProfile.saveProfile is false even if initialConnectionProfile.saveProfile is true, reset the flag here.
-			connectionProfile.saveProfile = initialConnectionProfile.saveProfile;
-			await this._connectionManagementService.connectAndSaveProfile(connectionProfile, undefined, {
-				saveTheConnection: isUndefinedOrNull(connectionCompletionOptions.saveConnection) ? true : connectionCompletionOptions.saveConnection,
-				showDashboard: isUndefinedOrNull(connectionCompletionOptions.showDashboard) ? false : connectionCompletionOptions.showDashboard,
-				params: undefined,
-				showConnectionDialogOnError: isUndefinedOrNull(connectionCompletionOptions.showConnectionDialogOnError) ? true : connectionCompletionOptions.showConnectionDialogOnError,
-				showFirewallRuleOnError: isUndefinedOrNull(connectionCompletionOptions.showFirewallRuleOnError) ? true : connectionCompletionOptions.showFirewallRuleOnError
-			});
+			if (connectionCompletionOptions.saveConnection) {
+				await this._connectionManagementService.connectAndSaveProfile(connectionProfile, undefined, {
```

Comment:
You're changing behavior here - now if someone sets saveProfile on the initialConnectionProfile they won't actually end up saving the connection unless they also specify connectionCompletionOptions.saveConnection.

I think you're on the right track (save connection doesn't seem like it should belong in connectionProfile) but let's make sure we aren't breaking existing behavior here and ideally actually remove saveConnection from connectionProfile entirely.

Reply:
Added back the `connectionProfile.saveProfile = initialConnectionProfile.saveProfile;`

I agree that we should remove `saveProfile` from `ConnectionProfile`, but that change would take a lot of API changes, which I want to avoid for now until the Connection refactor.